src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -193,6 +193,48 @@ T dot(const SGVector<T>& a, const SGVector<T>& b)
 	return infer_backend(a, b)->dot(a, b);
 }
 
+/** Performs the operation C = A .* B where ".*" denotes elementwise multiplication.
+ *
+ * This version returns the result in a newly created matrix.
+ *
+ * @param A First matrix
+ * @param B Second matrix
+ * @return The result of the operation
+ */
+template <typename T>
+SGMatrix<T> element_prod(SGMatrix<T>& a, SGMatrix<T>& b)
+{
+	REQUIRE(a.num_rows == b.num_rows && a.num_cols == b.num_cols,
+			"Dimension mismatch! A(%d x %d) vs B(%d x %d)\n",
+			a.num_rows, a.num_cols, b.num_rows, b.num_cols);
+	return infer_backend(a, b)->element_prod(a, b);
+}
+
+/** Performs the operation C = A .* B where ".*" denotes elementwise multiplication.
+ *
+ * This version returns the result in-place.
```

Comment:
And also if/whether it is possible to pass one of the operands arguments (A or B) as a result? (since you do that in unit test)